public/index.js

```diff
@@ -1,23 +1,79 @@
 import "./build/restaurantList.js";
-import "./build/restaurantDetail.js";
+import "./build/restaurantPage.js";
+import "./build/header.js";
+import "./build/restaurantCard.js"
+import * as requests from "./requests.js"
 
-const rootElement = document.getElementById('root');
-const pageElement = document.createElement('main');
+const rootElement = document.getElementById("root");
+const pageElement = document.createElement("main");
 
 rootElement.appendChild(pageElement);
 
-const restaurants = [
-    { name: "Restaurant A" },
-    { name: "Restaurant B" },
-    { name: "Restaurant C" }
-];
+Handlebars.registerPartial("restaurantCard", Handlebars.templates["restaurantCard.hbs"]);
 
+function renderHeader() {
+    const template = Handlebars.templates["header.hbs"];
+    rootElement.insertAdjacentHTML("afterbegin", template());
 
-function renderRestaurantList() {
-    const template = Handlebars.templates['restaurantList.hbs'];
-    pageElement.innerHTML = template({ restaurantList: restaurants });
+    document.getElementsByClassName("logo")[0].addEventListener("click", () => {
+        renderRestaurantList();
+        window.location.pathname = "/";
```

Comment:
window.location.pathname вызывает перезагрузку страницы. Предлагаю заменить на history.pushState
```suggestion
        history.pushState({}, "", "/");
```